src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/PremisesController.kt

```diff
@@ -84,9 +84,11 @@ class PremisesController(
     val premises = premisesService.getPremises(premisesId)
       ?: throw NotFoundProblem(premisesId, "Premises")
 
+    val deliusPrincipal = httpAuthService.getDeliusPrincipalOrThrow().name
```

Comment:
Again, I feel like this might be better fetching the principal and then getting the name. Even though I know it'll throw if it doesn't work, chaining methods like this makes me a bit nervous. Also means that if we get some strange behaviour, it's harder to track down.